### PR TITLE
Add Rust-Toolchain, refactor generics, typos/cleanup

### PR DESCRIPTION
**Pull request by insipx**
Fixed a TODO that required `Default` on `XmtpApiClient`. 

I also generally did some refactoring of type parameters, although this is subjective, so I did not do it everywhere (mostly in the builder and ClientBuilder). I wanted to get thoughts on it. IMO, it's much easier to understand generics in code + documentation like `Widget<Frame, Foo, Crypto>` rather than the terse version `Widget<Q, N, O, X, P>`, especially as usage of generics increases. This does have the downside of occasionally breaking the line if Generic names are very long, or if there are lots and lots of generics, EX:

```
Widget<
      FrameForTheGpuWindow,
      CryptographicPrimitivesForY,
      GenericB,
      GenericX,
>
```

Although I've found this to be uncommon, and the benefit of readable generics to be really nice

Generally just some things I noticed, can bring over to #272 once merged if desired

### CODE REVIEW

File: rust-toolchain.toml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[toolchain]
+channel = "stable"
```
Comment (insipx) [inline rust-toolchain.toml:2]: We can also pin to a specific version, not sure what the preference is

Comment (neekolas) [reply rust-toolchain.toml:2]: `stable` works for me. We have enough crazy build-time stuff happening in the bindings, last thing we need is to be doing that on a nightly build

**Comment by neekolas** [comment]
Thank you for tackling this @insipx. These look like great quality of life improvements. Things like readable generics will definitely be helpful as we onboard more devs to this repo.

It's some really unfortunate timing on this one. A few days ago, it wasn't clear at all that we were going to be sunsetting this crate and copying the code into a new one. But here we are. Unfortunately, you're going to duplicate these changes on the `mls` branch. Apologies for not announcing the move to a new crate earlier to save you some duplicate work.

**Review by richardhuaaa — APPROVED** [review]
Thank you for doing this cleanup!! I anticipate us also moving more of your changes across over time as the new crate becomes fleshed out, so I think it will be useful either way.

**Comment by insipx** [comment]
doing changes to MLS in new PR to keep things small

**Pull request merged**